framework/src/main/java/dev/ikm/komet/framework/view/ViewMenuTask.java

```diff
@@ -560,9 +560,11 @@ private static void addChangeItemsForView(ViewCalculator viewCalculator, List<Me
             Platform.runLater(() -> {
                 MenuItem sourceMenu = (MenuItem) event.getSource();
                 Menu parentMenu = sourceMenu.getParentMenu();
-                parentMenu.getItems().clear();
-                TinkExecutor.threadPool().execute(TaskWrapper.make(new ViewMenuTask(viewCalculator, observableView),
-                        (List<MenuItem> result) -> parentMenu.getItems().addAll(result)));
+                if (parentMenu != null) {
+                    parentMenu.getItems().clear();
+                    TinkExecutor.threadPool().execute(TaskWrapper.make(new ViewMenuTask(viewCalculator, observableView),
+                            (List<MenuItem> result) -> parentMenu.getItems().addAll(result)));
```

Comment:
Is this menu already attached to a Scene? If so we'd want to use Platform.runLater to execute the code that changes the UI in the UI thread

Reply:
I am late getting these messages. It's too late in the game. But next time small stuff like this feel free to fix as a shared branch.
I agree with @dholubek  this is read only for the demo and not to be invoked (trigger actions). But I believe the code was really lifted from classic Komet. Ideally when using a thread pool it provides activity information in the Activities tab for long running loads (building many menu options). 

I think @dukke  is correct that the parent (context menu) is attached to a Scene, and when adding to it (parentmenu) should be done on the JavaFX thread.

Reply:
@carldea sounds good. Do you want me to add this to the ticket we mentioned?